src/main/java/io/getunleash/repository/FeatureRepositoryImpl.java

```diff
@@ -0,0 +1,219 @@
+        return () -> {
+            if (throttler.performAction()) {
+                try {
+                    ClientFeaturesResponse response = featureFetcher.fetchFeatures();
+                    eventDispatcher.dispatch(response);
+                    if (response.getStatus() == ClientFeaturesResponse.Status.CHANGED) {
+                        String clientFeatures = response.getClientFeatures().get();
+
+                        this.engine.takeState(clientFeatures);
+                        featureBackupHandler.write(clientFeatures);
+                    } else if (response.getStatus() == ClientFeaturesResponse.Status.UNAVAILABLE) {
+                        if (!ready && unleashConfig.isSynchronousFetchOnInitialisation()) {
+                            throw new UnleashException(
+                                    String.format(
+                                            "Could not initialize Unleash, got response code %d",
+                                            response.getHttpStatusCode()),
+                                    null);
+                        }
+                        if (ready) {
+                            throttler.handleHttpErrorCodes(response.getHttpStatusCode());
+                        }
+                        return;
+                    }
+                    throttler.decrementFailureCountAndResetSkips();
+                    if (!ready) {
+                        eventDispatcher.dispatch(new UnleashReady());
+                        ready = true;
+                    }
+                } catch (UnleashException e) {
+                    handler.accept(e);
+                } catch (YggdrasilInvalidInputException | YggdrasilError e) {
+                    handler.accept(new UnleashException("Error when fetching features", e));
+                }
+            } else {
+                throttler.skipped(); // We didn't do anything this iteration, just reduce the count
+            }
+        };
+    }
+
+    public Integer getFailures() {
+        return this.throttler.getFailures();
+    }
+
+    public Integer getSkips() {
+        return this.throttler.getSkips();
+    }
+
+    @Override
+    public Boolean isEnabled(String toggleName, UnleashContext context) {
+        try {
+            return this.engine.isEnabled(toggleName, YggdrasilAdapters.adapt(context));
+        } catch (YggdrasilInvalidInputException | YggdrasilError e) {
+            LOGGER.error("Error when checking feature toggle {}", toggleName, e);
+            return null;
+        }
+    }
+
+    @Override
+    public Optional<VariantDef> getVariant(String toggleName, UnleashContext context) {
+        try {
+            return Optional.ofNullable(
+                    this.engine.getVariant(toggleName, YggdrasilAdapters.adapt(context)));
+        } catch (YggdrasilInvalidInputException | YggdrasilError e) {
+            LOGGER.error("Error when checking feature toggle {}", toggleName, e);
+            return null;
```

Comment:
Similar comment as above
```suggestion
            return Optional.empty();
```
Or maybe default variant